storage/store.go

```diff
@@ -1280,11 +1283,16 @@ func raftEntryFormatter(data []byte) string {
 // updateStoreStatus updates the store's status proto.
 func (s *Store) updateStoreStatus() {
 	now := s.ctx.Clock.Now().WallTime
-	s.status.UpdatedAt = now
-	s.status.RangeCount = int32(len(s.ranges))
+	status := &proto.StoreStatus{
+		StoreID:    s.Ident.StoreID,
+		NodeID:     s.Ident.NodeID,
+		UpdatedAt:  now,
+		RangeCount: int32(len(s.ranges)),
+		Stats:      proto.MVCCStats(s.scanner.Stats().MVCC),
+	}
+	atomic.StorePointer(&s.status, unsafe.Pointer(status))
```

Comment:
You could avoid the existing race by initializing `Store.status` in `Store.Start` before the range scanner is started. Probably good to do that even if you stick with the atomic pointer stuff because `Store.Start` should really not be stomping on the status if range scanner has updated it.